### ISSUE TITLE
Exception throws by Console and expand/collapse do not works

### ISSUE THREAD

**Issue by rat-moonshine** [prominic/Moonshine-IDE#286]
Pulling first updates after #214 and on first run of the application, compiler thrown following error:

```
TypeError: Error #1009: Cannot access a property or method of a null object reference.
	at actionScripts.plugin.console.view::ConsoleViewTextArea/appendtext()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/view/ConsoleViewTextArea.as:105]
	at actionScripts.plugin.console::ConsolePlugin/consoleOutputHandler()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsolePlugin.as:286]
	at flash.events::EventDispatcher/dispatchEventFunction()
	at flash.events::EventDispatcher/dispatchEvent()
	at actionScripts.plugin.console::ConsoleOutputter$/outputMsg()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:91]
	at actionScripts.plugin.console::ConsoleOutputter$/formatOutput()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:86]
	at actionScripts.plugin.console::ConsoleOutputter/notice()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:47]
	at actionScripts.plugin.console::ConsolePlugin/aboutCommand()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsolePlugin.as:358]
	at Function/<anonymous>()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsolePlugin.as:138]
	at Function/http://adobe.com/AS3/2006/builtin::apply()
	at SetIntervalTimer/onTimer()
	at flash.utils::Timer/_timerDispatch()
	at flash.utils::Timer/tick()
TypeError: Error #1009: Cannot access a property or method of a null object reference.
	at actionScripts.plugin.console.view::ConsoleViewTextArea/appendtext()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/view/ConsoleViewTextArea.as:105]
	at actionScripts.plugin.console::ConsolePlugin/consoleOutputHandler()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsolePlugin.as:286]
	at flash.events::EventDispatcher/dispatchEventFunction()
	at flash.events::EventDispatcher/dispatchEvent()
	at actionScripts.plugin.console::ConsoleOutputter$/outputMsg()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:91]
	at actionScripts.plugin.console::ConsoleOutputter$/formatOutput()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:86]
	at actionScripts.plugin.console::ConsoleOutputter/error()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:52]
	at actionScripts.plugin.errors::UncaughtErrorsPlugin/uncaughtErrorHandler()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/errors/UncaughtErrorsPlugin.as:76]
	at flash.events::EventDispatcher/dispatchEventFunction()
	at flash.events::EventDispatcher/dispatchEvent()
	at actionScripts.plugin.console::ConsoleOutputter$/outputMsg()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:91]
	at actionScripts.plugin.console::ConsoleOutputter$/formatOutput()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:86]
	at actionScripts.plugin.console::ConsoleOutputter/notice()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsoleOutputter.as:47]
	at actionScripts.plugin.console::ConsolePlugin/aboutCommand()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsolePlugin.as:358]
	at Function/<anonymous>()[/Users/rat-moonshine/ide/MoonshineSharedCore/src/actionScripts/plugin/console/ConsolePlugin.as:138]
	at Function/http://adobe.com/AS3/2006/builtin::apply()
	at SetIntervalTimer/onTimer()
	at flash.utils::Timer/_timerDispatch()
	at flash.utils::Timer/tick()
```

It looked like that some exception were thrown by this line: https://github.com/prominic/Moonshine-IDE/blob/master/ide/MoonshineSharedCore/src/actionScripts/plugin/console/view/ConsoleViewTextArea.as#L105.

After restarting Moonshine I couldn't able to get the error anymore, but I see console always starts collapsed and clicking on its expand-arrow icon does nothing.

**Comment by JoelProminic** [comment]
I also noticed that the console was collapsed by default.  Once I expanded it manually once, it seemed to reopen properly.

On a related note, the dragable area for the console is very thin.
![image](https://user-images.githubusercontent.com/24532731/40682976-c77e3800-6352-11e8-97bc-9cf618bc24f4.png)

I would expect to able to click and drag on the light grey bar, but that does not work.  I need to click and drag from just above that bar (highlighted in green in the screenshot).

Similarly, I found that the clickable area for expand/collapse is just above the dark grey button with an arrow.  I highlighed the approximate location of the area in blue.

**Comment by piotrzarzycki21** [comment]
I did make sure that Console view is created before we write on anything, so issue should be fixed.

In case of divider for the console I made it a bit higher, but it's require lot of more work. Maybe we should switch completely to spark DividedGroup which seems to be more pleasant to changes. I have raised separate issue for that.

**Comment by JoelProminic** [comment]
The divider separator is a lot easier to select now on macOS.   I'll update this issue if I noticed that the console is collapsed by default again.

**Comment by rat-moonshine** [comment]
I test this with latest updates from the Git, as a fresh startup. For this I've removed "C:\Users\[user]\AppData\Roaming\idToBeReplacedByANT" from my system. Moonshine started, this didn't thrown any exception this time. The console started collapsed, and not expanded even if I clicked the arrow (in the dragbar) but only gets a natural height if I drag the bar to a certain height. 

![image](https://user-images.githubusercontent.com/24370870/40821534-265f215c-6585-11e8-86fb-6b4d6ef759c5.png)

Per original or at least existing design (unless we chose to ignore the last design), a fresh startup should always expand the console by default - or at least the expand/collapse button should work expectedly. This may needs to check.

**Comment by piotrzarzycki21** [comment]
I have made couple of improvements. Let's see whether it helps. Please test it on your sight.

**Comment by JoelProminic** [comment]
I tested this on macOS:
- I confirmed that  after clearing the local settings directory, the console is open by default on the first launch
- The user-modified console height is preserved when restarting the application later
- The console is easy to expand and collapse
